src/server/scripts/Spells/spell_priest.cpp

```diff
@@ -323,6 +328,78 @@ class spell_pri_divine_hymn : public SpellScript
     }
 };
 
+// 110744 - Divine Star
+struct areatrigger_pri_divine_star : AreaTriggerAI
+{
+    areatrigger_pri_divine_star(AreaTrigger* areatrigger) : AreaTriggerAI(areatrigger) {}
+
+    void OnUpdate(uint32 diff) override
+    {
+        _scheduler.Update(diff);
+    }
+
+    void OnUnitEnter(Unit* unit) override
+    {
+        if (Unit* caster = at->GetCaster())
+        {
+            if (std::find(_affectedUnits.begin(), _affectedUnits.end(), unit->GetGUID()) == _affectedUnits.end())
+            {
+                if (caster->IsValidAttackTarget(unit))
+                    caster->CastSpell(unit, SPELL_PRIEST_DIVINE_STAR_DAMAGE, CastSpellExtraArgs(TriggerCastFlags(TRIGGERED_IGNORE_GCD | TRIGGERED_IGNORE_CAST_IN_PROGRESS)));
+                else if (caster->IsValidAssistTarget(unit))
+                    caster->CastSpell(unit, SPELL_PRIEST_DIVINE_STAR_HEAL, CastSpellExtraArgs(TriggerCastFlags(TRIGGERED_IGNORE_GCD | TRIGGERED_IGNORE_CAST_IN_PROGRESS)));
+
+                _affectedUnits.push_back(unit->GetGUID());
+            }
+        }
+    }
+
+    void OnDestinationReached() override
+    {
+        if (Unit* caster = at->GetCaster())
+        {
+            _affectedUnits.clear();
+
+            if (!_returningToCaster)
+            {
+                _scheduler.Schedule(Milliseconds(), [this, caster](TaskContext task)
+                {
+                    if (!_returningToCaster || caster->GetDistance(_casterCurrentPosition) > 1.0f)
+                    {
+                        _returningToCaster = true;
+                        _casterCurrentPosition = caster->GetPosition();
+
+                        Movement::PointsArray returnSplinePoints;
+
+                        for (uint8 i = 0; i < 4; i++)
+                        {
+                            G3D::Vector3 returnPoint;
+                            returnPoint.x = (i < 2) ? at->GetPositionX() : caster->GetPositionX();
```

Comment:
Any particular reason why these are 4 return points and the first two being the same and also the last two being the same?

Reply:
On retail, it seems to be 4 different points at which the spline redirects onto the caster when returning.

Reply:
So 4 different points, but right now you have 4 points of which half and half are the same

Reply:
![image](https://user-images.githubusercontent.com/30064812/178994453-4fc2af7a-2b79-48af-8f35-f7de0ac32b12.png)
on retail first two points are duplicates and last two points are duplicates as well; first two being the AT's pos and last two being the caster's pos.